qemu/tests/fio_perf.py

```diff
@@ -0,0 +1,224 @@
+import logging
+import os
+import commands
+import threading
+import re
+import time
+from autotest.client import utils
+from autotest.client.shared import error
+from virttest import utils_misc, utils_test
+
+
+def format_result(result, base="12", fbase="2"):
+    """
+    Format the result to a fixed length string.
+
+    :param result: result need to convert
+    :param base: the length of converted string
+    :param fbase: the decimal digit for float
+    """
+    if isinstance(result, str):
+        value = "%" + base + "s"
+    elif isinstance(result, int):
+        value = "%" + base + "d"
+    elif isinstance(result, float):
+        value = "%" + base + "." + fbase + "f"
+    return value % result
+
+
+def check_status(session, timeout, num):
+    """
+    Output disk info including disk status
+
+    :param session: VM session
+    :param timeout: Timeout in seconds
+    :param num: The number of data disks
+    """
+
+    disk_status_cmd = "echo list disk > cmd && echo exit >>"
+    disk_status_cmd += " cmd && diskpart /s cmd"
+    disks = []
+    start_time = time.time()
+    while time.time() - start_time < timeout:
+        disks_str = session.cmd(disk_status_cmd)
+        logging.info("disks_str is %s" % disks_str)
+        if len(re.findall("Disk %s" % num, disks_str)) >= 1:
+            disks = re.findall("Disk %s.*\n" % num, disks_str)
+            break
+    return disks
+
+
+def get_version(session, result_file, guest_ver_cmd, type, driver_format, timeout):
+    kvm_ver = utils.system_output("rpm -qa qemu-kvm-rhev && rpm -qa qemu-kvm")
+    host_ver = os.uname()[2]
+
+    result_file.write("### kvm-userspace-ver : %s\n" % kvm_ver)
+    result_file.write("### kvm_version : %s\n" % host_ver)
+
+    if driver_format != "ide":
+        result = session.cmd_output(guest_ver_cmd, timeout)
+        if type == "windows":
+            guest_ver = re.findall(".*?(\d{2}\.\d{2}\.\d{3}\.\d{4}).*?", result)
+            result_file.write("### guest-kernel-ver :Microsoft Windows [Version %s]\n" % guest_ver[0])
+        else:
+            result_file.write("### guest-kernel-ver :%s" % result)
+    else:
+        result_file.write("### guest-kernel-ver : Microsoft Windows [Version ide driver format]\n")
```

Comment:
No, actually we mainly test virtio_blk and virtio_scsi.  The second if is the scenario.